lib/routes/shared/security_pin/security_pin_page.dart

```diff
@@ -0,0 +1,172 @@
+import 'package:breez/bloc/blocs_provider.dart';
+import 'package:breez/bloc/user_profile/breez_user_model.dart';
+import 'package:breez/bloc/user_profile/security_model.dart';
+import 'package:breez/bloc/user_profile/user_actions.dart';
+import 'package:breez/bloc/user_profile/user_profile_bloc.dart';
+import 'package:breez/routes/shared/security_pin/prompt_pin_code.dart';
+import 'package:breez/routes/shared/security_pin/security_pin_warning_dialog.dart';
+import 'package:breez/theme_data.dart' as theme;
+import 'package:breez/widgets/back_button.dart' as backBtn;
+import 'package:breez/widgets/route.dart';
+import 'package:flutter/material.dart';
+
+class SecurityPage extends StatefulWidget {
+  SecurityPage({Key key}) : super(key: key);
+
+  @override
+  SecurityPageState createState() {
+    return SecurityPageState();
+  }
+}
+
+class SecurityPageState extends State<SecurityPage> {
+  UserProfileBloc _userProfileBloc;
+
+  bool _isInit = false;
+
+  @override
+  void initState() {
+    super.initState();
+  }
+
+  @override
+  void didChangeDependencies() {
+    super.didChangeDependencies();
+    if (!_isInit) {
+      _userProfileBloc = AppBlocsProvider.of<UserProfileBloc>(context);
+      _isInit = true;
+    }
+  }
+
+  void _updateSecurityModel(SecurityModel securityModel, {String pinCode, bool secureBackupWithPin, bool delete = false}) {
```

Comment:
Why also delete as parameter? we can just pass the right values instead.

Reply:
I agree this is not a good solution, I'm open to suggestions but this was the reasoning:

We decided to use single user action that updates security model (UpdateSecurityModel).
If a parameter is not provided to this function, we are setting it to it's value in the current security model.
To delete pinCode we pass it as null, which would we passed as it's current value to the UpdateSecurityModel user action if we don't set delete param to true.

Reply:
I am suggesting this: 
```suggestion
  void _updateSecurityModel(SecurityModel securityModel) {
     _userProfileBloc.userActionsSink.add(updateSecurityModelAction);
  }
```
Or even better call directly
`_userProfileBloc.userActionsSink.add(updateSecurityModelAction);` from the context it is needed